src/clj/puppetlabs/services/jruby/jruby_service.clj

```diff
@@ -20,12 +21,15 @@
                            [:ShutdownService shutdown-on-error]]
   (init
     [this context]
-    (let [config (core/initialize-config (get-config))
+    (let [initial-config (get-config)
           service-id (tk-services/service-id this)
-          agent-shutdown-fn (partial shutdown-on-error service-id)]
+          agent-shutdown-fn (partial shutdown-on-error service-id)
+          config (core/initialize-config (assoc-in initial-config
+                                                   [:jruby :lifecycle :shutdown-on-error]
+                                                   agent-shutdown-fn))]
       (core/verify-config-found! config)
```

Comment:
This is somewhat unrelated to this PR, but it seems like with the location of where this line of code is, it will never do anything? The initial commit comes from https://github.com/puppetlabs/puppetserver/pull/123/commits/0ef0afd9ecb00c8cc558a2efb1b894463e1a3ecc, but there's no test and I don't think there's any way that `initialize-config` can return nil or an empty map, which is what this function checks for).